anycloud/node/src/datastore.ts

```diff
@@ -34,31 +34,35 @@ export class DataStore {
     if (!dsKey) return undefined;
     if (this.isLocal) {
       if (dsKey in this.localDS) {
-        const val = this.localDS[dsKey]; 
-        if (typeof(val) === 'object') {
-          return JSON.stringify(val);
-        } else {
-          return `${val}`;
-        }
+        const val = this.localDS[dsKey];
+        return JSON.stringify(val);
       } else {
         return undefined;
       }
     }
     try {
-      return await this.request('GET', `${this.ctrlPortUrl}get/${dsKey.toString()}`);
+      return JSON.stringify(await this.request('GET', `${this.ctrlPortUrl}get/${dsKey.toString()}`));
     } catch (e) {
       return e;
     }
   }
 
   async set(dsKey: string, dsValue: any): Promise<string> {
     if (!dsKey) return 'fail';
+    const parsedDSValue = (function () {
```

Comment:
This name is off. This is the `stringifiedDSValue`, right?

Reply:
Also, absorbing the error like this may not be the best idea, because *why* the set failed is unclear.

I think having this be something like `Promise<Error | undefined>` might be a better choice? You'd still need to try-catch it, but now the end users can check if they got an error object, something like:

```js
async set(dsKey: string, dsValue: any): Promise<Error | undefined> {
  if (!dsKey) return new Error("No key was provided");
  let stringifiedDSValue;
  try {
    stringifiedDSValue = JSON.stringify(dsValue);
  } catch (e) {
    return e;
  }
  if (this.isLocal) { ...
```

Reply:
I've remove the `null` assignment because it made no sense. What I have done is to save the value even though it is no JSON serializable, so in the catch just assign the incoming value. Because is is a `key:value` datastore, I think we should not make it only accept JSON serializable objects. Nevertheless, your suggestion of returning an `Error` object makes sense and we must do something like that. Not sure if `Promise<Error | undefined>`, I think maybe  Promise<Error | boolean> makes more sense following your other comment below.